### PR TITLE
fix: Systems can be added twice because of missing check by type

### PR DESCRIPTION
**Pull request by Dan-Crane**
## Description

Added a check when adding a system that is already in [`_systemsByType`](https://github.com/Dan-Crane/oxygen/blob/main/lib/src/system/system_manager.dart#L13)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`). 
- [ ] The dart analyzer (`dart analyze`) does not report any problems on my PR.
- [x] I read and followed the [Effective Dart Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I am willing to follow-up on review comments in a timely manner.
- [x] I am happy with the current version of this PR and it is ready to be reviewed
- [ ] If the PR still has the `Draft` status, remove it by clicking on the `Ready for review` button in this PR.

## Breaking Change

Does your PR require Oxygen users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR*

[issue database]: https://github.com/flame-engine/oxygen/issues
[Contributor Guide]: https://github.com/flame-engine/oxygen/blob/master/CONTRIBUTING.md
[Effective Dart Style Guide]: https://dart.dev/guides/language/effective-dart/style
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning

### CODE REVIEW

File: test/system/system_test.dart [hunk 1]
```diff
@@ -42,23 +42,48 @@ class SystemD extends System {
...
-        ..registerSystem(systemB)
-        ..registerSystem(systemC)
-        ..registerSystem(systemD)
-        ..init();
-
-      expect(
-        world.systemManager.systems,
-        equals([systemC, systemB, systemD, systemA]),
-      );
+    group('registerSystem -', () {
```
Comment (wolfenrain) [inline test/system/system_test.dart]:
```suggestion
    group('registerSystem', () {
```

File: test/system/system_test.dart [hunk 2]
```diff
@@ -42,23 +42,48 @@ class SystemD extends System {
...
+      });
+
+      test(
+          'The "system.world == null assertion" should occur when adding a '
+          'system with an initialized world', () {
+        final system = SystemA();
+        final world = World()..registerSystem(system);
+
+        expect(
+          () => world..registerSystem(system),
```
Comment (wolfenrain) [inline test/system/system_test.dart]: We dont need to use the `..` operator here right?

**Review by wolfenrain — APPROVED** [review]
Thanks for the PR! Left a new nits but all in all looks fine to me!

File: test/system/system_test.dart (hunk 2 above)
Comment (Dan-Crane) [reply test/system/system_test.dart]: yes, you are right) thanks

**Review by wolfenrain — APPROVED** [review]

**Review by spydon — APPROVED** [review]

**Comment by wolfenrain** [comment]
@Dan-Crane are you able to follow-up on this to get the workflows unblocked?

**Comment by Dan-Crane** [comment]
@wolfenrain  hi!) 
Do i understand correctly, i need to correct what CICD is swearing at?

**Pull request merged**